pkg/controller/events/event_forwarder.go

```diff
@@ -0,0 +1,93 @@
+package events
+
+import (
+	"context"
+	"strings"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+type EventHandler struct {
+	client.Client
+	Scheme *runtime.Scheme
+}
+
+func (e *EventHandler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	log := ctrl.LoggerFrom(ctx)
+
+	event := &corev1.Event{}
+	if err := e.Get(ctx, req.NamespacedName, event); err != nil {
+		return ctrl.Result{}, err
+	}
+
+	obj := &unstructured.Unstructured{}
+	obj.SetGroupVersionKind(schema.FromAPIVersionAndKind(event.InvolvedObject.APIVersion, event.InvolvedObject.Kind))
+	if err := e.Get(ctx, types.NamespacedName{Name: event.InvolvedObject.Name, Namespace: event.InvolvedObject.Namespace}, obj); err != nil {
+		return ctrl.Result{}, err
+	}
+	ef := obj.GetAnnotations()["vshn.net/forward-events-to"]
+	if ef != "" {
+		log.Info("Forwarding event", "event", event.Name, "to", ef)
+		knc := strings.Split(ef, "/")
+		a := knc[0] + "/" + knc[1]
+		k := knc[2]
+		ns := knc[3]
+		c := knc[4]
+
+		claim := &unstructured.Unstructured{}
+		claim.SetGroupVersionKind(schema.FromAPIVersionAndKind(a, k))
+		if err := e.Get(ctx, types.NamespacedName{Name: c, Namespace: ns}, claim); err != nil {
+			return ctrl.Result{}, err
+		}
+		nev := &corev1.Event{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      event.Name,
+				Namespace: ns,
+			},
+			InvolvedObject: corev1.ObjectReference{
+				Kind:            k,
+				Namespace:       ns,
+				Name:            c,
+				APIVersion:      a,
+				ResourceVersion: claim.GetResourceVersion(),
+				UID:             claim.GetUID(),
+			},
+		}
+		_, err := controllerutil.CreateOrPatch(ctx, e.Client, nev, func() error {
```

Comment:
I'm not sure if patching existing events makes sense here. Won't old events pup-up at the top again in such a case?

Reply:
The idea with patching is, that we can leverage the two event properties: `firstTimestamp` and  `lastTimestamp`. With those two fields it's clear when it started and when it happened the last time.
This is only possible when patching the existing event. Otherwise, the user won't know if the event is actually still triggering or not.

Reply:
I was under the impression that each event is a distinct object and those fields are computed. But I'm not sure if that's really the case.

Does the update of the timestamp and counts work as expected with this solution? If yes it's fine, but if something is not working as intended, then we probably should use the `EventRecorder`, which will handle those details for us.

Reply:
As discussed, we'll use this solution for now, if we find any issues, we'll change to the recorder.